lib/core/src/display/shape/text/text_field/cursor.rs

```diff
@@ -211,11 +227,64 @@ impl<'a> CursorNavigation<'a> {
         next_line.map(|line| self.near_same_x_in_another_line(position,line))
     }
 
+    fn next_word_position(&mut self, position:&TextLocation) -> Option<TextLocation> {
+        let line  = position.line;
+        let words = Words::new(self.content.lines()[line].chars());
+        let word  = words.iter().find(|word| {
+            let (word_start, _) = word[0];
+            let word_end        = word_start + word.len();
+            position.column < word_end
+        }).map(|word| {
+            word[0].0 + word.len()
+        });
+        let next_line    = Some(line + 1).filter(|l| *l < self.content.lines().len());
+        match (word, next_line) {
+            (Some(column), _)  => Some(TextLocation{line,column}),
+            (None, Some(line)) => Some(TextLocation::at_line_begin(line)),
+            (None, None)       => None
+        }
+    }
+
+    fn prev_word_position(&mut self, position:&TextLocation) -> Option<TextLocation> {
+        let line  = position.line;
+        let words = Words::new(self.content.lines()[line].chars());
+        let word  = words.iter().rev().find(|word| {
+            let (word_start, _) = word[0];
+            position.column > word_start
+        }).map(|word| {
+            word[0].0
+        });
+        let previous_line = Some(line).filter(|l| *l > 0).map(|l| l - 1);
+        match (word, previous_line) {
+            (Some(column), _)  => Some(TextLocation{line,column}),
+            (None, Some(line)) => Some(TextLocation{line,column:self.content.lines()[line].len()}),
+            (None, None)       => None
+        }
```

Comment:
This is a code duplication as well, please refactor it to something more generic.